### ISSUE TITLE
Auto-refresh + Wordpress multi-site

### ISSUE THREAD

**Issue by bdkjones** [bdkjones/CodeKit2#309]
I am developing WordPress themes on a multisite (to test different content sets and themes). Since it's subdomain based, I can't toggle between the sites and maintain the auto-refresh.

A domain whitelist/blacklist feature (as in CodeKit 1, except project based) would work much better in the case of multiple domains or sites sharing a codebase.

While awesome, I don't need super fancy "refresh all my browsers on every device" all the time, but I would like my 4 Chrome tabs to refresh when I save a file.

**Comment by imperator99** [comment]
Hi Bryan,

I'm having a similar issue with a Drupal multisite installation (managed through Aegir). My development site would be something like http://[sitename].[domain].[host].com.au, so the theme files will be served from http://[sitename].[domain].[host].com.au/sites/[sitename].[domain].[host].com.au/themes/[theme-name]/css/[theme-name].styles.css.

When using the CodeKit external server, it replaces both instances of my site name with the local server, let's say [local]:5757. So it starts looking for my CSS files (and scripts, and images) in http://[local]:5757/sites/[local]:5757/themes/[theme-name]/_. This gives me a long list of 404 errors in the browser (not to mention an entirely unstyled site, and no refreshing), because the files are actually in http://[local]:5757/sites/[sitename].[domain].[host].com.au/themes/[theme-name]/css/[theme-name]/_. I can navigate to any of these files in the CodeKit server by replacing the second local server with the name of my development server.

I'd love some way of sorting this out, it is the currently the only thing holding me back on certain setups. I just need to be able to tell CodeKit to only replace that first instance of the server name, not all of them.

Cheers,
Jesse.

**Comment by bdkjones** [comment]
Are you using Vhosts for your external server? What name have you assigned to the virtual host?

**Comment by imperator99** [comment]
I don't know - I don't have control over the environment itself. Because it is all done through Aegir, such things would be automatically assigned based on whatever I elected to call the site when setting it up. CodeKit is evidently expecting files to be in a particular case and the folder structure I'm presenting it with is giving it grief.

**Comment by bdkjones** [comment]
Hmm... I could potentially fix this, but I have to do some performance testing to make sure it's viable. Replacing all instances of string 'xyz' is very fast. But if I write code that looks for only the first occurrence of 'xyz' within either quotes or double quotes after an "href=" block, now we're slowing things down.

**Comment by imperator99** [comment]
Yeah that makes sense and I can see why you would do it the way it works now - but I can potentially see a lot of opportunity for use in our government set up and being able to use Aegir along with CodeKit would be fantastic, as this is the direction a lot of Australian government agencies are heading now.

**Comment by weskoop** [comment]
I currently have *.dev pointing to a vagrant box running WP Multisite (sub-domain, matching up with a production server)

I'm using this for wildcard DNS: http://serverfault.com/a/118589
dnsmasq.conf:
`address=/dev/192.168.50.4`

But I'm still using CodeKit 1, it's the only way that I can maintain refreshing. Also having Codekit Fiddling with the domains and auto replacing domains makes it really hard to test when where is an error, as it creates another layer of complexity.

Tweaks here may also positively affect: https://github.com/bdkjones/codekit2/issues/318
